Platform/TigerlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -682,7 +683,16 @@ ClearSmi (
   // Clear the status before setting smi enable
   //
   SmiSts = IoRead32 ((UINTN)(UINT32)(ACPI_BASE_ADDRESS + R_ACPI_IO_SMI_STS));
-  Pm1Sts = IoRead16 ((UINTN)(ACPI_BASE_ADDRESS + R_ACPI_IO_PM1_STS));
+  Pm1Sts = IoRead32 ((UINTN)(ACPI_BASE_ADDRESS + R_ACPI_IO_PM1_STS));
+  Pm1Cnt = IoRead16 ((UINTN)(ACPI_BASE_ADDRESS + R_ACPI_IO_PM1_CNT));
+
+  // Clear RTC Alarm only if wake source is RTC SMI#
+  if (((Pm1Sts & (B_ACPI_IO_PM1_STS_RTC_EN | B_ACPI_IO_PM1_STS_RTC)) ==
+       (B_ACPI_IO_PM1_STS_RTC_EN | B_ACPI_IO_PM1_STS_RTC)) &&
+     !(Pm1Cnt & B_ACPI_IO_PM1_CNT_SCI_EN)) {
+    IoWrite8 (R_RTC_IO_INDEX, R_RTC_IO_REGC);
+    (void)IoRead8 (R_RTC_IO_TARGET);

```

Comment:
This clears Alarm Flag in RTC Reg C only if above conditions are met.
But, B_ACPI_IO_PM1_STS_RTC is cleared all the time by line#713. I think the B_ACPI_IO_PM1_STS_RTC is also cleared only when the conditions are met like RTC Alarm Flag.

Reply:
Reasonable, thanks. it will be updated.

Reply:
Could you double check if the S3 hang issue was fixed by commit 791d7a0bebf479b already?

Reply:
The previous commit does not fix the RTC S3 hang issue addressed by this patch.

Reply:
> Is there a place where GlobalSmi can be set before RestoreS3Regs()? We clear some EN bits as early as GetPlatformPowerState () in Stage1BBoardInitLib.c to avoid generating SMI on normal boot also.
> 
> If there is no concern like above, Looks good to me.

Reply:
I see. It means AF flag has to be cleared to avoid SMI is triggered again.